tests/test_tokenization_common.py

```diff
@@ -310,6 +310,31 @@ def convert_batch_encode_plus_format_to_encode_plus(batch_encode_plus_sequences)
             for i in range(len(batch_encode_plus_sequences["input_ids"]))
         ]
 
+    # TODO: this test can be combined with `test_sentencepiece_tokenize_and_convert_tokens_to_string` after the latter is extended to all tokenizers.
+    def test_tokenize_special_tokens(self):
+        """Test `tokenize` with special tokens."""
+        tokenizers = [self.get_tokenizer(do_lower_case=True)] if self.test_slow_tokenizer else []
+        for tokenizer in tokenizers:
+            with self.subTest(f"{tokenizer.__class__.__name__}"):
+                SPECIAL_TOKEN_1 = "[SPECIAL_TOKEN_1]"
+                SPECIAL_TOKEN_2 = "[SPECIAL_TOKEN_2]"
+
+                # TODO: Can we replace `unique_no_split_tokens` and `all_special_tokens` with one variable(property) for a better maintainability?
+                # The list in which `_add_tokens` method stores special tokens. (in tokenization_utils.py)
+                tokenizer.unique_no_split_tokens = [SPECIAL_TOKEN_1]
+                # The property storing additional special tokens which occur in `all_special_tokens`. (in tokenization_utils_base.py)
+                tokenizer.additional_special_tokens = [SPECIAL_TOKEN_2]
+                # Add all special tokens into `unique_no_split_tokens`.
+                tokenizer.sanitize_special_tokens()
```

Comment:
How about leveraging the `tokenizer.add_special_tokens` method so that this follows a user workflow?

Reply:
I got your point. However, `tokenizer.add_special_tokens` method is only appropriate to test tokens expected to exist in `additional_special_token`. The method #13483 used is directly calling `tokenizer.add_tokens(token, special_tokens=True)` to add new special tokens, and the tokens added through this method will never occur in `tokenizer.all_special_tokens`. That's why we need this fix and the test.

TL;DR, I think we can use `tokenizer.add_special_tokens` to test whether special tokens stored in `tokenizer.all_special_tokens` can properly be tokenized, and, use `tokenizer.add_tokens` to test whether special tokens stored in `tokenizer.unique_no_split_tokens` can properly be tokenized.

What do you think?

Reply:
Yes that makes sense